### ISSUE TITLE
recover_orphan does not update `enqueued_at`

### ISSUE THREAD

**Issue by tmandke** [sidekiq/sidekiq#5985]
Ruby version: 3.0.4p208
Rails version: 6.1.7.3
Sidekiq / Pro / Enterprise version(s): sidekiq-pro 7.1.3

From time to time our queue latency pops and drops. We realized this was because of the jobs that get put back on the queue after being recovered by `recover_orphan` have their `enqueued_at` set to the time when the job was originally enqueued and not when they were recovered.

**Comment by mperham** [comment]
Correct. This is the trade off since recovery is done atomically within Lua, I see it as too risky and CPU intensive to parse the JSON, update the value and serialize it back. Instead I'd suggest you treat it as a sign that your app has a bug which orphans jobs and fix that bug instead.

**Comment by tmandke** [comment]
Completely agree that this is something we need to track down and fix. And great its a conscious trade off. 👍🏽
